src/components/Menu.vue

```diff
@@ -7,18 +7,17 @@
     </ion-header>
     <ion-content>
       <ion-list>
-        <ion-item v-if="hasPermission('APP_SELECT_PRODUCT_VIEW')" button @click="closeMenu(); router.push('/select-product')">
-          <ion-icon :icon="optionsOutline" slot="start" />
-          <ion-label>{{ $t("Create Rule") }}</ion-label>
-        </ion-item>
-        <ion-item v-if="hasPermission('APP_THRESHOLD_UPDATES_VIEW')" button @click="closeMenu(); router.push('/threshold-updates')">
-          <ion-icon :icon="pulseOutline" slot="start" />
-          <ion-label>{{ $t("Rule Pipeline") }}</ion-label>
-        </ion-item>
-        <ion-item button @click="closeMenu(); router.push('/settings')">
-          <ion-icon :icon="settingsOutline" slot="start" />
-          <ion-label>{{ $t("Settings") }}</ion-label>
-        </ion-item>
+        <ion-menu-toggle auto-hide="false" v-for="(p,i) in getValidMenuItems(appPages)" :key="i">
```

Comment:
The variables in v-for needs to be explanatory. Don't use character such as p and i.